commonmark-ext-styles/src/main/java/org/commonmark/ext/styles/internal/StylesDelimiterProcessor.java

```diff
@@ -30,22 +37,95 @@ public int getDelimiterUse(DelimiterRun opener, DelimiterRun closer) {
 
     @Override
     public void process(Text opener, Text closer, int delimiterCount) {
-        StringBuilder styleString = new StringBuilder();
+        // Check if the attributes can be applied - if the previous node is an Image, and if all the attributes are in
+        // the set of SUPPORTED_STYLES
+        if (opener.getPrevious() instanceof Image) {
+            boolean canApply = true;
+            List<Node> toUnlink = new ArrayList<>();
+
+            StringBuilder styleString = new StringBuilder();
+            Node tmp = opener.getNext();
+            while (tmp != null && tmp != closer) {
+                Node next = tmp.getNext();
+                // Only Text nodes can be used for attributes
+                if (tmp instanceof Text) {
+                    String styles = ((Text) tmp).getLiteral();
+                    for (String s : styles.split("\\s+")) {
+                        String[] attribute = s.split("=");
+                        // Check if the attribute is in SUPPORTED_STYLES.
+                        if (SUPPORTED_STYLES.contains(attribute[0].toLowerCase())) {
+                            styleString.append(((Text) tmp).getLiteral());
+                            // The tmp node can be unlinked, as we have retrieved its value.
+                            toUnlink.add(tmp);
+                        } else {
+                            // This attribute is not supported, so break here (no need to check any further ones).
+                            canApply = false;
+                            break;
+                        }
+                    }
+                } else {
+                    // This node type is not supported, so break here (no need to check any further ones).
+                    canApply = false;
+                    break;
+                }
+                tmp = next;
+            }
+
+            // Only if all of the above checks pass can the attributes be applied.
+            if (canApply) {
+                // Unlink the tmp nodes
+                for (Node node : toUnlink) {
+                    node.unlink();
+                }
+
+                if (styleString.length() > 0) {
+                    Styles styles = new Styles(styleString.toString());
+
+                    // The styles node is added as a child of the node to which the styles apply.
+                    Node nodeToStyle = opener.getPrevious();
+                    nodeToStyle.appendChild(styles);
+                }
+                return;
+            }
+        }
+
+        // If we got here then the attributes cannot be applied, so fallback to leaving the text unchanged.
+        // Need to add back the opening and closing characters (which are removed elsewhere).
```

Comment:
The opening and closing characters get removed elsewhere, so we need to add them back here.